File: backend/service/src/main/kotlin/com/teven/service/report/ReportService.kt

package com.teven.service.report

import com.teven.api.model.report.InventoryUsageReportResponse
import com.teven.api.model.report.StaffHoursReportRequest
import com.teven.api.model.report.StaffHoursReportResponse
import com.teven.data.report.ReportDao

class ReportService(private val reportDao: ReportDao) {
    suspend fun getStaffHoursReport(staffHoursReportRequest: StaffHoursReportRequest): List<StaffHoursReportResponse> {
        return reportDao.getStaffHoursReport(staffHoursReportRequest)
    }

    suspend fun getInventoryUsageReport(): List<InventoryUsageReportResponse> {
        return reportDao.getInventoryUsageReport()
    }
}

File: frontend/src/api/ReportService.ts
import type { StaffHoursReportRequest, StaffHoursReportResponse, InventoryUsageReportResponse } from '../types/reports';
import { apiClient } from './apiClient';

export class ReportService {
  static async getStaffHoursReport(request: StaffHoursReportRequest): Promise<StaffHoursReportResponse[]> {
    const queryParams = new URLSearchParams({
      startDate: request.startDate,
      endDate: request.endDate,
    }).toString();
    return apiClient<StaffHoursReportResponse[]>(`/api/reports/staff_hours?${queryParams}`);
  }

  static async getInventoryUsageReport(): Promise<InventoryUsageReportResponse[]> {
    return apiClient<InventoryUsageReportResponse[]>('/api/reports/inventory_usage');
  }
}
